### ISSUE TITLE
yaml files of a certain structure produce incorrectly generated code.

### ISSUE THREAD

**Issue by schoonc** [intlify/bundle-tools#32]
### Reporting a bug?

In the repository at the link there is an `en.yaml` file (it looks a bit weird, but that's because I tried to keep it as simple as possible to catch the bug). After performing a yarn build with this `en.yaml`, the build will not work (see screenshots below).

Note that with this `en.yaml` the build is broken:
![image](https://user-images.githubusercontent.com/4659633/125695655-926a9a9a-7af7-41ce-9c5b-b8778ac681d2.png)

But with this `en.yaml` (for example) the build is not broken:
![image](https://user-images.githubusercontent.com/4659633/125695667-04bdf025-5052-42af-ba53-15fcd2d0e25e.png)

### Expected behavior

Working build. There should be no errors in the console and this page should appear:
![image](https://user-images.githubusercontent.com/4659633/125695969-52a2f7fb-4344-4e19-95a6-5050eb02a82c.png)

### Reproduction

https://github.com/schoonc/intlify--vue-i18n-next--issues--575

1. Clone the repository.
2. `$ yarn`
3. `$ yarn build`
4. Open `index.html` in browser.
5. See the console.

### System Info

```shell
System:
    OS: Linux 5.11 Ubuntu 21.04 (Hirsute Hippo)
    CPU: (8) x64 AMD Ryzen 5 3550H with Radeon Vega Mobile Gfx
    Memory: 4.73 GB / 15.50 GB
    Container: Yes
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 14.17.0 - /usr/bin/node
    Yarn: 1.22.5 - /usr/bin/yarn
    npm: 7.18.1 - /usr/bin/npm
  Browsers:
    Chrome: 91.0.4472.77
    Firefox: 89.0.2
  npmPackages:
    vue: ^3.0.0 => 3.1.4 
    vue-i18n: 9.1.7 => 9.1.7
```

### Screenshot

![image](https://user-images.githubusercontent.com/4659633/125694175-769e09c2-6737-49e8-9495-49fc5d70abeb.png)
![image](https://user-images.githubusercontent.com/4659633/125694242-6247276a-31fb-427d-94dc-c67369398832.png)

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n-next/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/intlify/vue-i18n-next/discussions)

**Comment by schoonc** [comment]
Yes, it looks very much like it's because of the [literal style](https://yaml.org/spec/1.2/spec.html#id2795688). By the way, the problem does not occur in development mode (`yarn serve`).

**Comment by PeterAlfredLee** [comment]
@kazupon 
It seems this issue is related to the source map that generated by `vue-i18n-loader` after parses the YAML file.
I have tested, if we replace this [line](https://github.com/intlify/bundle-tools/blob/1cd0c58dea9c7b8b04f39e903a88f8cf984b4973/packages/vue-i18n-loader/src/index.ts#L41) from 
```js
this.callback(null, code, map)
```
to
```js
this.callback(null, code, null)
```
this problem won't happen.

This is what I found so far, any ideas?

**Comment by kazupon** [comment]
@PeterAlfredLee 
thanks for your help!

I’ve checked same happen.

I did some research on the source map generated by the generator. Unfortunately, I couldn't figure out why. 😞 

I wondered if it made sense to support source maps again.

In general, source map support makes it possible to identify source code with devtools such as chrome.

At the moment, vue-i18n-loader does support it, so you can break point to files like yaml other than js as shown in the screen capture.

However, even if you can breakpoint yaml files, you cannot check the state of the variables like you can with javascript. Currently, you can only see the call stack.

![image](https://user-images.githubusercontent.com/72989/128729480-11ca5de6-62d0-4697-8ee1-352375d28e60.png)

So, I am beginning to think that source map support is not very useful.

To avoid this problem, I want to add a `sourcemap` option to the vue-i18n-loader.
The `sourcemap` option is disabled by default, `false`. If you set it to `true`, the behavior is to generate a source map to pass to webpack, as in the current version.

Th sourcemap generation is not generated by default,  so it’s breaking change.

What do you think about this?

**Comment by PeterAlfredLee** [comment]
> The sourcemap option is disabled by default, false. If you set it to true, the behavior is to generate a source map to pass to webpack, as in the current version.

Good point. But I have no idea how to just disable the `sourcemap` of `vue-i18n-loader`. I only found the global option `productionSourceMap` in vue-cli documents.

**Comment by PeterAlfredLee** [comment]
> for vue-i18n-loader, I like the same specification as the productionSourceMap option as well as the Vue CLI.
https://cli.vuejs.org/config/#productionsourcemap

Just opened PR #39. Please check it.

**Comment by kazupon** [comment]
I've just released v3.0.0
please check it!
https://github.com/intlify/bundle-tools/releases/tag/vue-i18n-loader%403.0.0

**Comment by schoonc** [comment]
It works! Thank you!
